pkg/gameserversets/controller_test.go

```diff
@@ -157,6 +157,44 @@ func TestComputeReconciliationAction(t *testing.T) {
 			wantNumServersToAdd: 0,
 			wantIsPartial:       true,
 		},
+		{
+			desc: "DeletingUnhealthyGameServers",
```

Comment:
@jkowalski added these extra tests, which resulted in the bug fix above - wanted to do more testing for shutdown scenarios, to try and catch anything that would cause that leaky `Unhealthy` gameservers to rear its head.

Any issues with these tests? And are there other scenarios you can think of that we should test that I didn't cover?

Reply:
LGTM